lib/puppetserver/ca/action/prune.rb

```diff
@@ -50,26 +53,37 @@ def run(inputs)
 
           # Getting the CRL(s)
           loader = X509Loader.new(puppet.settings[:cacert], puppet.settings[:cakey], puppet.settings[:cacrl])
+          inventory_file = puppet.settings[:cert_inventory]
+          cadir = puppet.settings[:cadir]
 
           verified_crls = loader.crls.select { |crl| crl.verify(loader.key) }
+          number_of_removed_duplicates = 0
+          number_of_removed_crl_entries = 0
 
           if verified_crls.length == 1
             puppet_crl = verified_crls.first
             @logger.inform("Total number of certificates found in Puppet's CRL is: #{puppet_crl.revoked.length}.")
-            number_of_removed_duplicates = prune_CRL(puppet_crl)
 
-            if number_of_removed_duplicates > 0
+            if serialnumbers
```

Comment:
It'd be nice if you could pass in both.  Maybe we can make this a case statement instead?

Reply:
Good point. I was thinking user want to use one or other at a time. I can update that.

Reply:
Updated